src/main/java/org/takes/rq/RqMultipart.java

```diff
@@ -350,4 +357,121 @@ public InputStream body() throws IOException {
+        public Fake(final String... dispositions) throws IOException {
+            this(
+                new Request() {
+                    @Override
+                    public Iterable<String> head() throws IOException {
+                        return Arrays.asList(
+                            "POST /h?u=3 HTTP/1.1",
+                            "Host: www.example.com",
+                            // @checkstyle LineLengthCheck (1 line)
+                            "Content-Type: multipart/form-data; boundary=AaB02x",
+                            "Content-Length: 100001"
+                        );
+                    }
+                    @Override
+                    public InputStream body() throws IOException {
+                        return new ByteArrayInputStream(new byte[]{});
+                    }
+                },
+                dispositions
+            );
+        }
+        @Override
+        public Iterable<Request> part(final CharSequence name) {
+            return this.fake.part(name);
+        }
+        @Override
+        public Iterable<String> names() {
+            return this.fake.names();
+        }
+        @Override
+        public Iterable<String> head() throws IOException {
+            return this.fake.head();
+        }
+        @Override
+        public InputStream body() throws IOException {
+            return this.fake.body();
+        }
+        /**
+         * Fake body creator.
+         * @param dispositions Fake request body parts
+         * @return InputStream of given dispositions
+         */
+        private InputStream fakeBody(final String... dispositions) {
+            final String boundary = "AaB02x";
+            final Collection<String> parts = new LinkedList<String>();
+            for (final String disposition : dispositions) {
+                parts.add(String.format("--%s", boundary));
+                parts.add(disposition);
+            }
+            parts.add("Content-Transfer-Encoding: utf-8");
+            parts.add("");
+            parts.add("the start\r\t\n\u20ac\n\n\n\t\r\t\n\n\n\r\n the end");
+            parts.add(String.format("--%s--", boundary));
+            return new ByteArrayInputStream(
+                    RqMultipart.Fake.join(parts.iterator()).getBytes()
+            );
+        }
+        /**
+         * Iterable<T> to List<T> converter.
+         * @param source Source Iterable
+         * @return List of given source
+         */
+        private static List<String> convert(final Iterable<String> source) {
+            final List<String> destination = new LinkedList<String>();
+            for (final String each : source) {
+                destination.add(each);
+            }
+            return destination;
+        }
+        /**
+         * CRLF joiner.
+         * @param iter Source iterator to join by CRLF
+         * @return String of iterator joined with CRLF
+         */
+        private static String join(final Iterator<String> iter) {
```

Comment:
@erimerturk do we really need this method? seems like we are re-inventing the wheel...